fix: FIT-991: OOM on CSV async_export

matt-bernstein opened the pull request:
followup to https://github.com/HumanSignal/label-studio/pull/5612 fixing last site where this occurs

**NOTE: this is _not_ necessarily the root cause of the related TRIAG bug; that still has to be established. It's something I noticed along the way.

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

matt-bernstein committed:
fix: FIT-991: OOM on CSV async_export

bmartel commented on label_studio/data_export/mixins.py:
```diff
@@ -338,8 +341,8 @@ def convert_file(self, to_format, download_resources=False, hostname=None):
 
-            with open(input_file_path, 'wb') as file_:
-                file_.write(self.file.open().read())
+            with open(input_file_path, 'wb') as dst, self.file.open('rb') as src:
```
Do these changes prevent from allocating the complete file in memory?

matt-bernstein replied:
Yep, `open()` just returns a file handler, `.read()` was the culprit that loaded it into memory
https://til.simonwillison.net/python/copy-file

matt-bernstein committed:
make tempdir work for both sync and async download

matt-bernstein committed:
Merge remote-tracking branch 'origin/develop' into fb-FIT-991/dont-export-in-memory

matt-bernstein commented:
/fmt
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/19344903041)

matt-bernstein committed:
Apply pre-commit linters

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/19344903041

robot-ci-heartex commented:
This PR is stale because it has been open 45 days with no activity. Remove `stale` label or comment or this will be closed in 10 days.

robot-ci-heartex commented:
This PR was closed because it has been stalled for 10 days with no activity.
